src/NovelRT/Ecs/Input/InputSystem.cpp

```diff
@@ -0,0 +1,88 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root
+// for more information.
+
+#include <NovelRT/Ecs/Ecs.h>
+
+namespace NovelRT::Ecs::Input
+{
+    InputSystem::InputSystem(std::shared_ptr<PluginManagement::IWindowingPluginProvider> windowingProvider, std::shared_ptr<PluginManagement::IInputPluginProvider> inputProvider)
+    {
+        _actionMap = std::map<NovelRT::Atom, std::string>();
+        _firstUpdate = true;
+        _service = inputProvider->GetInputService();
+        _service->Initialise(windowingProvider->GetWindowingDevice()->GetHandle());
+    }
+
+    void InputSystem::Update(Timing::Timestamp delta, Ecs::Catalogue catalogue)
+    {
+        if (_firstUpdate)
+        {
+            auto actions = _service->GetAllMappings();
+            for (auto act : actions)
```

Comment:
Could we make this `auto&&` instead of `auto`? Or is it faster to copy by value?

Reply:
Done